lib/zuora/objects/payment.rb

```diff
@@ -1,41 +1,41 @@
-    #validates_date_of :settled_on, :allow_nil => true
-    #validates_length_of :soft_descriptor, :maximum => 35, :allow_nil => true
-    #validates_length_of :soft_descriptor_phone, :maximum => 20, :allow_nil => true
-    #validates_inclusion_of :status, :in => %w(Canceled Draft Error Posted Processing Processed Voided)
-    #validates_date_of :submitted_on, :allow_nil => true
-    #validates_inclusion_of :transferred_to_accouning, :allow_nil => true, :in => %w(Processing Yes Error Ignore)
-    #validates_inclusion_of :type, :in => %w(External Electronic)
-    #validates_length_of :updated_by_id, :maximum => 32
-    #validates_datetime_of :updated_date
+    validates_length_of :accounting_code, :maximum => 100, :allow_nil => true
+    validates_numericality_of :amount
+    validates_numericality_of :applied_credit_balance_amount
+    validates_numericality_of :applied_invoice_amount
+    validates_length_of :auth_transaction_id, :maximum => 50, :allow_nil => true
+    validates_length_of :bank_identification_number, :maximum => 6, :allow_nil => true
+    # validates_date_of :cancelled_on, :allow_nil => true
+    validates_length_of :comment, :maximum => 255, :allow_nil => true
+    validates_length_of :created_by_id, :maximum => 32, :allow_nil => true
+    # validates_datetime_of :effective_date
+    validates_length_of :gateway_order_id, :maximum => 70, :allow_nil => true
+    validates_inclusion_of :gateway_state, :in => %w(NotSubmitted Submitted Settled MarkedForSubmission)
+    # validates_datetime_of :marked_for_submission_on, :allow_nil => true
+    validates_length_of :payment_number, :maximum => 32, :allow_nil => true
+    validates_length_of :reference_id, :maximum => 30, :allow_nil => true
+    validates_numericality_of :refund_amount, :allow_nil => true
+    validates_length_of :second_payment_reference_id, :maximum => 60, :allow_nil => true
+    # validates_date_of :settled_on, :allow_nil => true
+    validates_length_of :soft_descriptor, :maximum => 35, :allow_nil => true
+    validates_length_of :soft_descriptor_phone, :maximum => 20, :allow_nil => true
+    validates_inclusion_of :status, :in => %w(Canceled Draft Error Posted Processing Processed Voided)
+    # validates_date_of :submitted_on, :allow_nil => true
+    validates_inclusion_of :transferred_to_accouning, :allow_nil => true, :in => %w(Processing Yes Error Ignore)
+    validates_inclusion_of :type, :in => %w(External Electronic)
+    validates_length_of :updated_by_id, :maximum => 32
+    # validates_datetime_of :updated_date
 
-    #define_attributes do
-      #read_only :bank_identification_number, :created_by_id, :created_date, :gateway_response,
-        #:gateway_response_code, :updated_by_id, :updated_date
-    #end
-  #end
-#end
+    define_attributes do
+      read_only :bank_identification_number, :created_by_id, :created_date, :gateway_response,
+        :gateway_response_code, :updated_by_id, :updated_date
+    end
+  end
+end
```

Comment:
Do we know why this class is uncommented when it wasn't previously?

Reply:
Part of the original import from upstream as seen [here](https://github.com/Sage/zuora/commit/35a6399545bd685b5054f1adcd401721e7c35190#diff-8f88ca3e66f063d39f293812a97a9dbf68ceca26a4d592d528f882b6b443e8a0).